failed to upload chart to  chartmuseum, NullPointerException

**Environment (plugin version, maven version, OS, ...)**:

System: Mac 10.15
Maven: 3.6.1
plugin: 4.12


**What happened**:

upload chart failed.

```xml
    <build>
        <plugins>
            <plugin>
                <groupId>com.kiwigrid</groupId>
                <artifactId>helm-maven-plugin</artifactId>
                <version>${plugin.maven.helm.version}</version>
                <configuration>
                    <chartDirectory>${project.basedir}</chartDirectory>
                    <chartVersion>${project.version}</chartVersion>
                    <helmDownloadUrl>https://mirrors.huaweicloud.com/helm/${helm.version}/helm-${helm.version}-darwin-amd64.tar.gz</helmDownloadUrl>
                    <helmHomeDirectory>${project.basedir}/target/helm/home</helmHomeDirectory>
                    <uploadRepoStable>
                        <name>stable-repo</name>
                        <url>****</url>
                        <type>CHARTMUSEUM</type>
                        <username>****</username>
                        <password>****</password>
                    </uploadRepoStable>
                    <uploadRepoSnapshot>
                        <name>snapshot-repo</name>
                        <url>*****</url>
                        <type>CHARTMUSEUM</type>
                        <username>****</username>
                        <password>****</password>
                    </uploadRepoSnapshot>
                </configuration>
                <executions>
                    <execution>
                        <id>deploy</id>
                        <phase>deploy</phase>
                        <goals>
                            <goal>init</goal>
                            <goal>lint</goal>
                            <goal>package</goal>
                            <goal>upload</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
        </plugins>
```
**What you expected to happen**:

```
Apache Maven 3.6.2 (40f52333136460af0dc0d7232c0dc0bcf0d9e117; 2019-08-27T23:06:16+08:00)
Maven home: /usr/local/Cellar/maven/3.6.2/libexec
Java version: 11.0.4, vendor: AdoptOpenJDK, runtime: /Library/Java/JavaVirtualMachines/adoptopenjdk-11.jdk/Contents/Home
Default locale: zh_CN_#Hans, platform encoding: UTF-8
OS name: "mac os x", version: "10.15", arch: "x86_64", family: "mac"
[DEBUG] Created new class realm maven.api
[DEBUG] Importing foreign packages into class realm maven.api
[DEBUG]   Imported: javax.annotation.* < plexus.core
[DEBUG]   Imported: javax.annotation.security.* < plexus.core
[DEBUG]   Imported: javax.enterprise.inject.* < plexus.core
[DEBUG]   Imported: javax.enterprise.util.* < plexus.core
[DEBUG]   Imported: javax.inject.* < plexus.core
[DEBUG]   Imported: org.apache.maven.* < plexus.core
[DEBUG]   Imported: org.apache.maven.artifact < plexus.core
[DEBUG]   Imported: org.apache.maven.classrealm < plexus.core
[DEBUG]   Imported: org.apache.maven.cli < plexus.core
[DEBUG]   Imported: org.apache.maven.configuration < plexus.core
[DEBUG]   Imported: org.apache.maven.exception < plexus.core
[DEBUG]   Imported: org.apache.maven.execution < plexus.core
[DEBUG]   Imported: org.apache.maven.execution.scope < plexus.core
[DEBUG]   Imported: org.apache.maven.lifecycle < plexus.core
[DEBUG]   Imported: org.apache.maven.model < plexus.core
[DEBUG]   Imported: org.apache.maven.monitor < plexus.core
[DEBUG]   Imported: org.apache.maven.plugin < plexus.core
[DEBUG]   Imported: org.apache.maven.profiles < plexus.core
[DEBUG]   Imported: org.apache.maven.project < plexus.core
[DEBUG]   Imported: org.apache.maven.reporting < plexus.core
[DEBUG]   Imported: org.apache.maven.repository < plexus.core
[DEBUG]   Imported: org.apache.maven.rtinfo < plexus.core
[DEBUG]   Imported: org.apache.maven.settings < plexus.core
[DEBUG]   Imported: org.apache.maven.toolchain < plexus.core
[DEBUG]   Imported: org.apache.maven.usability < plexus.core
[DEBUG]   Imported: org.apache.maven.wagon.* < plexus.core
[DEBUG]   Imported: org.apache.maven.wagon.authentication < plexus.core
[DEBUG]   Imported: org.apache.maven.wagon.authorization < plexus.core
[DEBUG]   Imported: org.apache.maven.wagon.events < plexus.core
[DEBUG]   Imported: org.apache.maven.wagon.observers < plexus.core
[DEBUG]   Imported: org.apache.maven.wagon.proxy < plexus.core
[DEBUG]   Imported: org.apache.maven.wagon.repository < plexus.core
[DEBUG]   Imported: org.apache.maven.wagon.resource < plexus.core
[DEBUG]   Imported: org.codehaus.classworlds < plexus.core
[DEBUG]   Imported: org.codehaus.plexus.* < plexus.core
[DEBUG]   Imported: org.codehaus.plexus.classworlds < plexus.core
[DEBUG]   Imported: org.codehaus.plexus.component < plexus.core
[DEBUG]   Imported: org.codehaus.plexus.configuration < plexus.core
[DEBUG]   Imported: org.codehaus.plexus.container < plexus.core
[DEBUG]   Imported: org.codehaus.plexus.context < plexus.core
[DEBUG]   Imported: org.codehaus.plexus.lifecycle < plexus.core
[DEBUG]   Imported: org.codehaus.plexus.logging < plexus.core
[DEBUG]   Imported: org.codehaus.plexus.personality < plexus.core
[DEBUG]   Imported: org.codehaus.plexus.util.xml.Xpp3Dom < plexus.core
[DEBUG]   Imported: org.codehaus.plexus.util.xml.pull.XmlPullParser < plexus.core
[DEBUG]   Imported: org.codehaus.plexus.util.xml.pull.XmlPullParserException < plexus.core
[DEBUG]   Imported: org.codehaus.plexus.util.xml.pull.XmlSerializer < plexus.core
[DEBUG]   Imported: org.eclipse.aether.* < plexus.core
[DEBUG]   Imported: org.eclipse.aether.artifact < plexus.core
[DEBUG]   Imported: org.eclipse.aether.collection < plexus.core
[DEBUG]   Imported: org.eclipse.aether.deployment < plexus.core
[DEBUG]   Imported: org.eclipse.aether.graph < plexus.core
[DEBUG]   Imported: org.eclipse.aether.impl < plexus.core
[DEBUG]   Imported: org.eclipse.aether.installation < plexus.core
[DEBUG]   Imported: org.eclipse.aether.internal.impl < plexus.core
[DEBUG]   Imported: org.eclipse.aether.metadata < plexus.core
[DEBUG]   Imported: org.eclipse.aether.repository < plexus.core
[DEBUG]   Imported: org.eclipse.aether.resolution < plexus.core
[DEBUG]   Imported: org.eclipse.aether.spi < plexus.core
[DEBUG]   Imported: org.eclipse.aether.transfer < plexus.core
[DEBUG]   Imported: org.eclipse.aether.version < plexus.core
[DEBUG]   Imported: org.fusesource.jansi.* < plexus.core
[DEBUG]   Imported: org.slf4j.* < plexus.core
[DEBUG]   Imported: org.slf4j.event.* < plexus.core
[DEBUG]   Imported: org.slf4j.helpers.* < plexus.core
[DEBUG]   Imported: org.slf4j.spi.* < plexus.core
[DEBUG] Populating class realm maven.api
[INFO] Error stacktraces are turned on.
[DEBUG] Message scheme: color
[DEBUG] Message styles: debug info warning error success failure strong mojo project
[DEBUG] Reading global settings from /usr/local/Cellar/maven/3.6.2/libexec/conf/settings.xml
[DEBUG] Reading user settings from /Users/x/.m2/settings.xml
[DEBUG] Reading global toolchains from /usr/local/Cellar/maven/3.6.2/libexec/conf/toolchains.xml
[DEBUG] Reading user toolchains from /Users/x/.m2/toolchains.xml
[DEBUG] Using local repository at /Users/x/.m2/repository
[DEBUG] Using manager EnhancedLocalRepositoryManager with priority 10.0 for /Users/x/.m2/repository
[INFO] Scanning for projects...
[DEBUG] Extension realms for project com.sunmaoyun.framework:mars-boot-container-helm:pom:1.1.0-SNAPSHOT: (none)
[DEBUG] Looking up lifecycle mappings for packaging pom from ClassRealm[plexus.core, parent: null]
[DEBUG] Extension realms for project com.sunmaoyun.framework:mars-boot-container:pom:1.1.0-SNAPSHOT: (none)
[DEBUG] Looking up lifecycle mappings for packaging pom from ClassRealm[plexus.core, parent: null]
[DEBUG] Extension realms for project com.sunmaoyun.framework:mars-boot:pom:1.1.0-SNAPSHOT: (none)
[DEBUG] Looking up lifecycle mappings for packaging pom from ClassRealm[plexus.core, parent: null]
[DEBUG] Resolving plugin prefix helm from [org.apache.maven.plugins, org.codehaus.mojo]
[DEBUG] Resolved plugin prefix helm to com.kiwigrid:helm-maven-plugin from POM com.sunmaoyun.framework:mars-boot-container-helm:pom:1.1.0-SNAPSHOT
[DEBUG] === REACTOR BUILD PLAN ================================================
[DEBUG] Project: com.sunmaoyun.framework:mars-boot-container-helm:pom:1.1.0-SNAPSHOT
[DEBUG] Tasks:   [helm:upload]
[DEBUG] Style:   Regular
[DEBUG] =======================================================================
[INFO] 
[INFO] ----------< com.sunmaoyun.framework:mars-boot-container-helm >----------
[INFO] Building mars-boot-container-helm 1.1.0-SNAPSHOT
[INFO] --------------------------------[ pom ]---------------------------------
[DEBUG] Resolving plugin prefix helm from [org.apache.maven.plugins, org.codehaus.mojo]
[DEBUG] Resolved plugin prefix helm to com.kiwigrid:helm-maven-plugin from POM com.sunmaoyun.framework:mars-boot-container-helm:pom:1.1.0-SNAPSHOT
[DEBUG] Lifecycle default -> [validate, initialize, generate-sources, process-sources, generate-resources, process-resources, compile, process-classes, generate-test-sources, process-test-sources, generate-test-resources, process-test-resources, test-compile, process-test-classes, test, prepare-package, package, pre-integration-test, integration-test, post-integration-test, verify, install, deploy]
[DEBUG] Lifecycle clean -> [pre-clean, clean, post-clean]
[DEBUG] Lifecycle site -> [pre-site, site, post-site, site-deploy]
[DEBUG] === PROJECT BUILD PLAN ================================================
[DEBUG] Project:       com.sunmaoyun.framework:mars-boot-container-helm:1.1.0-SNAPSHOT
[DEBUG] Dependencies (collect): []
[DEBUG] Dependencies (resolve): []
[DEBUG] Repositories (dependencies): [aliyun-public (https://maven.aliyun.com/repository/public, default, releases), aliyun-center (https://maven.aliyun.com/repository/central, default, releases), aliyun-jecnter (https://maven.aliyun.com/repository/jcenter, default, releases), aliyun-google (https://maven.aliyun.com/repository/google, default, releases), aliyun-spring (https://maven.aliyun.com/repository/spring, default, releases+snapshots), spring-milestones (https://repo.spring.io/milestone, default, releases+snapshots), huaweicloud (https://mirrors.huaweicloud.com/repository/maven, default, releases), central (https://repo.maven.apache.org/maven2, default, releases)]
[DEBUG] Repositories (plugins)     : [aliyun-public (https://maven.aliyun.com/repository/public, default, releases), aliyun-center (https://maven.aliyun.com/repository/central, default, releases), aliyun-jecnter (https://maven.aliyun.com/repository/jcenter, default, releases), aliyun-google (https://maven.aliyun.com/repository/google, default, releases), aliyun-gradle-plugin (https://maven.aliyun.com/repository/gradle-plugin, default, releases), aliyun-spring (https://maven.aliyun.com/repository/spring, default, releases+snapshots), aliyun-spring-plugin (https://maven.aliyun.com/repository/spring-plugin, default, releases+snapshots), spring-milestones (https://repo.spring.io/milestone, default, releases), huaweicloud (https://mirrors.huaweicloud.com/repository/maven, default, releases), central (https://repo.maven.apache.org/maven2, default, releases)]
[DEBUG] -----------------------------------------------------------------------
[DEBUG] Goal:          com.kiwigrid:helm-maven-plugin:4.12:upload (default-cli)
[DEBUG] Style:         Regular
[DEBUG] Configuration: <?xml version="1.0" encoding="UTF-8"?>
<configuration>
  <appVersion>${helm.appVersion}</appVersion>
  <autoDetectLocalHelmBinary default-value="true">${helm.autoDetectLocalHelmBinary}</autoDetectLocalHelmBinary>
  <chartDirectory>/Users/x/Develop/Project/Me/Java/mars-boot/mars-boot-container/mars-boot-container-helm</chartDirectory>
  <chartVersion>1.1.0-SNAPSHOT</chartVersion>
  <excludes>${helm.excludes}</excludes>
  <helmDownloadUrl>https://mirrors.huaweicloud.com/helm/v2.9.1/helm-v2.9.1-darwin-amd64.tar.gz</helmDownloadUrl>
  <helmExecutableDirectory default-value="${project.build.directory}/helm">${helm.executableDirectory}</helmExecutableDirectory>
  <helmExtraRepos>${helm.extraRepos}</helmExtraRepos>
  <helmHomeDirectory>/Users/*/Develop/Project/Me/Java/mars-boot/mars-boot-container/mars-boot-container-helm/target/helm/home</helmHomeDirectory>
  <helmSecurity default-value="~/.m2/settings-security.xml">${helm.security}</helmSecurity>
  <outputDirectory default-value="${project.build.directory}/helm/repo">${helm.outputDirectory}</outputDirectory>
  <settings default-value="${settings}"/>
  <skip default-value="false">${helm.skip}</skip>
  <skipUpload default-value="false">${helm.upload.skip}</skipUpload>
  <uploadRepoSnapshot>
    <name>snapshot-repo</name>
    <url>*</url>
    <type>CHARTMUSEUM</type>
    <username>*</username>
    <password>*</password>${helm.uploadRepo.snapshot}</uploadRepoSnapshot>
  <uploadRepoStable>
    <name>stable-repo</name>
    <url>*</url>
    <type>CHARTMUSEUM</type>
    <username>*</username>
    <password>*</password>${helm.uploadRepo.stable}</uploadRepoStable>
  <useLocalHelmBinary default-value="false">${helm.useLocalHelmBinary}</useLocalHelmBinary>
</configuration>
[DEBUG] =======================================================================
[INFO] 
[INFO] --- helm-maven-plugin:4.12:upload (default-cli) @ mars-boot-container-helm ---
[DEBUG] Dependency collection stats: {ConflictMarker.analyzeTime=696349, ConflictMarker.markTime=301093, ConflictMarker.nodeCount=24, ConflictIdSorter.graphTime=427215, ConflictIdSorter.topsortTime=322173, ConflictIdSorter.conflictIdCount=18, ConflictIdSorter.conflictIdCycleCount=0, ConflictResolver.totalTime=2789695, ConflictResolver.conflictItemCount=24, DefaultDependencyCollector.collectTime=119780227, DefaultDependencyCollector.transformTime=6305556}
[DEBUG] com.kiwigrid:helm-maven-plugin:jar:4.12
[DEBUG]    org.apache.maven:maven-plugin-api:jar:3.6.1:compile
[DEBUG]       org.apache.maven:maven-model:jar:3.6.1:compile
[DEBUG]       org.apache.maven:maven-artifact:jar:3.6.1:compile
[DEBUG]          org.apache.commons:commons-lang3:jar:3.8.1:compile
[DEBUG]       org.eclipse.sisu:org.eclipse.sisu.plexus:jar:0.3.3:compile
[DEBUG]          javax.enterprise:cdi-api:jar:1.0:compile
[DEBUG]             javax.annotation:jsr250-api:jar:1.0:compile
[DEBUG]             javax.inject:javax.inject:jar:1:compile
[DEBUG]          org.eclipse.sisu:org.eclipse.sisu.inject:jar:0.3.3:compile
[DEBUG]          org.codehaus.plexus:plexus-component-annotations:jar:1.5.5:compile
[DEBUG]       org.codehaus.plexus:plexus-utils:jar:3.2.0:compile
[DEBUG]       org.codehaus.plexus:plexus-classworlds:jar:2.6.0:compile
[DEBUG]    org.apache.maven:maven-settings:jar:3.6.1:compile
[DEBUG]    org.sonatype.plexus:plexus-sec-dispatcher:jar:1.4:compile
[DEBUG]       org.sonatype.plexus:plexus-cipher:jar:1.4:compile
[DEBUG]    org.apache.commons:commons-compress:jar:1.18:compile
[DEBUG]    commons-io:commons-io:jar:2.6:compile
[DEBUG] Created new class realm plugin>com.kiwigrid:helm-maven-plugin:4.12
[DEBUG] Importing foreign packages into class realm plugin>com.kiwigrid:helm-maven-plugin:4.12
[DEBUG]   Imported:  < maven.api
[DEBUG] Populating class realm plugin>com.kiwigrid:helm-maven-plugin:4.12
[DEBUG]   Included: com.kiwigrid:helm-maven-plugin:jar:4.12
[DEBUG]   Included: org.apache.commons:commons-lang3:jar:3.8.1
[DEBUG]   Included: javax.enterprise:cdi-api:jar:1.0
[DEBUG]   Included: org.eclipse.sisu:org.eclipse.sisu.inject:jar:0.3.3
[DEBUG]   Included: org.codehaus.plexus:plexus-component-annotations:jar:1.5.5
[DEBUG]   Included: org.codehaus.plexus:plexus-utils:jar:3.2.0
[DEBUG]   Included: org.sonatype.plexus:plexus-sec-dispatcher:jar:1.4
[DEBUG]   Included: org.sonatype.plexus:plexus-cipher:jar:1.4
[DEBUG]   Included: org.apache.commons:commons-compress:jar:1.18
[DEBUG]   Included: commons-io:commons-io:jar:2.6
[DEBUG] Configuring mojo com.kiwigrid:helm-maven-plugin:4.12:upload from plugin realm ClassRealm[plugin>com.kiwigrid:helm-maven-plugin:4.12, parent: jdk.internal.loader.ClassLoaders$AppClassLoader@4b85612c]
[DEBUG] Configuring mojo 'com.kiwigrid:helm-maven-plugin:4.12:upload' with basic configurator -->
[DEBUG]   (s) autoDetectLocalHelmBinary = true
[DEBUG]   (s) chartDirectory = /Users/x/Develop/Project/Me/Java/mars-boot/mars-boot-container/mars-boot-container-helm
[DEBUG]   (s) chartVersion = 1.1.0-SNAPSHOT
[DEBUG]   (s) excludes = []
[DEBUG]   (s) helmDownloadUrl = https://mirrors.huaweicloud.com/helm/v2.9.1/helm-v2.9.1-darwin-amd64.tar.gz
[DEBUG]   (s) helmExecutableDirectory = /Users/*/Develop/Project/Me/Java/mars-boot/mars-boot-container/mars-boot-container-helm/target/helm
[DEBUG]   (s) helmExtraRepos = []
[DEBUG]   (s) helmHomeDirectory = /Users/*/Develop/Project/Me/Java/mars-boot/mars-boot-container/mars-boot-container-helm/target/helm/home
[DEBUG]   (s) helmSecurity = ~/.m2/settings-security.xml
[DEBUG]   (s) outputDirectory = /Users/*/Develop/Project/Me/Java/mars-boot/mars-boot-container/mars-boot-container-helm/target/helm/repo
[DEBUG]   (f) settings = org.apache.maven.execution.SettingsAdapter@15ac59c2
[DEBUG]   (f) skip = false
[DEBUG]   (f) skipUpload = false
[DEBUG]   (s) name = snapshot-repo
[DEBUG]   (s) url = *
[DEBUG]   (s) type = CHARTMUSEUM
[DEBUG]   (s) username = *
[DEBUG]   (s) password = *
[DEBUG]   (s) uploadRepoSnapshot = [snapshot-repo / *]
[DEBUG]   (s) name = stable-repo
[DEBUG]   (s) url = *
[DEBUG]   (s) type = CHARTMUSEUM
[DEBUG]   (s) username = *
[DEBUG]   (s) password = *
[DEBUG]   (s) uploadRepoStable = [stable-repo / *]
[DEBUG]   (s) useLocalHelmBinary = false
[DEBUG] -- end configuration --
[INFO] Uploading to *

[INFO] Uploading /Users/*/Develop/Project/Me/Java/mars-boot/mars-boot-container/mars-boot-container-helm/target/helm/repo/marsboot-1.1.0-SNAPSHOT.tgz...
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.058 s
[INFO] Finished at: 2019-10-30T09:54:19+08:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.kiwigrid:helm-maven-plugin:4.12:upload (default-cli) on project mars-boot-container-helm: Execution default-cli of goal com.kiwigrid:helm-maven-plugin:4.12:upload failed.: NullPointerException -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.kiwigrid:helm-maven-plugin:4.12:upload (default-cli) on project mars-boot-container-helm: Execution default-cli of goal com.kiwigrid:helm-maven-plugin:4.12:upload failed.
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default-cli of goal com.kiwigrid:helm-maven-plugin:4.12:upload failed.
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:148)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: java.lang.NullPointerException
    at java.io.Reader.<init> (Reader.java:167)
    at java.io.InputStreamReader.<init> (InputStreamReader.java:113)
    at org.apache.commons.io.IOUtils.copy (IOUtils.java:2440)
    at org.apache.commons.io.IOUtils.toString (IOUtils.java:1084)
    at com.kiwigrid.helm.maven.plugin.UploadMojo.uploadSingle (UploadMojo.java:77)
    at com.kiwigrid.helm.maven.plugin.UploadMojo.execute (UploadMojo.java:43)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
[ERROR] 
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException

```




**How to reproduce it** (as minimally and precisely as possible):


**Anything else we need to know**:

--- a/src/main/java/com/kiwigrid/helm/maven/plugin/UploadMojo.java
+++ b/src/main/java/com/kiwigrid/helm/maven/plugin/UploadMojo.java
@@ -49,7 +49,7 @@ public class UploadMojo extends AbstractHelmMojo {
 		}
 	}
 
-	private void uploadSingle(String file) throws IOException, BadUploadException, MojoExecutionException {
+	protected void uploadSingle(String file) throws IOException, BadUploadException, MojoExecutionException {
 		final File fileToUpload = new File(file);
 		final HelmRepository uploadRepo = getHelmUploadRepo();
 
@@ -74,11 +74,21 @@ public class UploadMojo extends AbstractHelmMojo {
 			IOUtils.copy(fileInputStream, connection.getOutputStream());
 		}
 		if (connection.getResponseCode() >= 300) {
-			String response = IOUtils.toString(connection.getErrorStream(), Charset.defaultCharset());
+			String response;
+			if (connection.getErrorStream() != null) {
+				response = IOUtils.toString(connection.getErrorStream(), Charset.defaultCharset());
+			} else if (connection.getInputStream() != null) {
+				response = IOUtils.toString(connection.getInputStream(), Charset.defaultCharset());
+			} else {
+				response = "No details provided";
+			}
 			throw new BadUploadException(response);
 		} else {
-			String response = IOUtils.toString(connection.getInputStream(), Charset.defaultCharset());
-			getLog().info(Integer.toString(connection.getResponseCode()) + " - " + response);
+			String message = Integer.toString(connection.getResponseCode());
+			if (connection.getInputStream() != null) {
+				message += " - " + IOUtils.toString(connection.getInputStream(), Charset.defaultCharset());
+			}
+			getLog().info(message);
 		}
 		connection.disconnect();
 	}

--- a/src/test/java/com/kiwigrid/helm/maven/plugin/UploadMojoTest.java
+++ b/src/test/java/com/kiwigrid/helm/maven/plugin/UploadMojoTest.java
@@ -4,15 +4,12 @@ import java.io.ByteArrayInputStream;
 import java.io.File;
 import java.io.IOException;
 import java.io.OutputStream;
-import java.net.Authenticator;
-import java.net.HttpURLConnection;
-import java.net.InetAddress;
-import java.net.PasswordAuthentication;
-import java.net.URL;
+import java.net.*;
 import java.nio.charset.StandardCharsets;
 import java.util.ArrayList;
 import java.util.List;
 
+import com.kiwigrid.helm.maven.plugin.exception.BadUploadException;
 import com.kiwigrid.helm.maven.plugin.junit.MojoExtension;
 import com.kiwigrid.helm.maven.plugin.junit.MojoProperty;
 import com.kiwigrid.helm.maven.plugin.junit.SystemPropertyExtension;
@@ -224,6 +221,35 @@ public class UploadMojoTest {
 		doReturn(tgzs).when(uploadMojo).getChartTgzs(anyString());
 
 		assertThrows(IllegalArgumentException.class, uploadMojo::execute, "Missing repo type must fail.");
+	}
+
+	@Test
+	public void verfifyNullErrorStreamOnFailedUpload(UploadMojo uploadMojo)
+			throws IOException, MojoExecutionException
+	{
+		final HelmRepository helmRepo = new HelmRepository();
+		helmRepo.setType(RepoType.CHARTMUSEUM);
+		helmRepo.setName("my-chartmuseum");
+		helmRepo.setUrl("https://somwhere.com/repo");
+		uploadMojo.setUploadRepoStable(helmRepo);
+
+		URL testChart = this.getClass().getResource("app-0.1.0.tgz");
+
+		final HttpURLConnection urlConnectionMock = Mockito.mock(HttpURLConnection.class);
+		doReturn(new NullOutputStream()).when(urlConnectionMock).getOutputStream();
+		doReturn(301).when(urlConnectionMock).getResponseCode();
+		doReturn(null).when(urlConnectionMock).getErrorStream();
+		doReturn(null).when(urlConnectionMock).getInputStream();
+		doNothing().when(urlConnectionMock).connect();
+		doReturn(urlConnectionMock).when(uploadMojo).getConnectionForUploadToChartmuseum();
+
+		try {
+			uploadMojo.uploadSingle(testChart.getFile());
+		} catch (BadUploadException e) {
+			assertNotNull(e.getMessage(), "Exception must provide a message");
+			return;
+		}
+		fail("BadUploadException expected on failed upload");
 	}
 
 	/** Writes to nowhere */

